core/include/moveit/task_constructor/stage.h

```diff
@@ -365,6 +365,7 @@ class Generator : public ComputeBase
 
 	virtual bool canCompute() const = 0;
 	virtual void compute() = 0;
+	void spawn(InterfaceState&& from, InterfaceState&& to, SubTrajectory&& trajectory);
```

Comment:
Please rename state arguments to `start` and `end`, which gives them a more concrete meaning.